### ISSUE TITLE
Unable to clear a previously set hint, SDL_SetHintWithPriority bails early

### ISSUE THREAD

**Issue by ephphatha** [libsdl-org/SDL#5120]
If `SDL_SetHintWithPriority` is called with a `NULL` value the check on line 55 bails early without attempting to set/clear the hint value.
https://github.com/libsdl-org/SDL/blob/505df1a99b033dc75d4a301f0fc2c5931b1098c3/src/SDL_hints.c#L55-L57

However later in the function value is checked again and if it was falsy (`NULL`) then the `hint->value` is set to `NULL`.
https://github.com/libsdl-org/SDL/blob/505df1a99b033dc75d4a301f0fc2c5931b1098c3/src/SDL_hints.c#L69-L77

This code is currently unreachable due to the earlier check. It looks like line 55 should just be `if (!name) {`.

This does still leave the hint in the linked list, should there be a way to remove a hint short of the nuclear option `SDL_ClearHints`?

**Comment by slouken** [comment]
Yes, there should. This needs more review to make sure NULL is handled properly and there are no race conditions if we decide to remove entries from the linked list.

**Comment by icculus** [comment]
Also, @slouken, can you sanity-check c391bdc1607de896a0f0b577e2b4a57e5f5740b7 for me?

This used to say "if the old or new hint is NULL or the hint value has changed, update the hint value" but that would unnecessarily call the callback if both values were NULL.

**Comment by slouken** [comment]
Remove the extra spaces, but the logic seems correct to me.

**Comment by icculus** [comment]
Done! 9703f2366347e16dac68ab688609a91099455285
